src/windows/video_widget.py

```diff
@@ -509,15 +509,23 @@ def paintEvent(self, event, *args):
     def centeredViewport(self, width, height):
         """ Calculate size of viewport to maintain aspect ratio """
 
-        window_size = QSizeF(width, height)
-        window_rect = QRectF(QPointF(0, 0), window_size)
```

Comment:
If I'm reading this right, here's the error. Window is always at the top left corner.

Reply:
You might be right.... but my brain struggles with this type of logic, lol

Reply:
That's not the error (I'm not sure what the error is, I didn't see what problem this was meant to address) — `window_rect` is the _window_ (the widget area being displayed in), of course its origin is in its own upper-left corner, and it  has the size indicated. You can't create a rect without an origin point, which is almost always `QPoint(0,0)` or `QPointF(0.0, 0.0)`.

This later call centers the viewport within the window:
```python
viewport_rect.moveCenter(window_rect.center())
```

Reply:
Well, anyway, like I said the lack of accounting for zoom was the only difference. Here's the entire change required to add zooming to my `centeredViewport()`:

```diff
diff --git a/src/windows/video_widget.py b/src/windows/video_widget.py
index a4d42969..5666a9a9 100644
--- a/src/windows/video_widget.py
+++ b/src/windows/video_widget.py
@@ -513,7 +513,9 @@ class VideoWidget(QWidget, updates.UpdateInterface):
         window_rect = QRectF(QPointF(0, 0), window_size)
 
         aspectRatio = self.aspect_ratio.ToFloat() * self.pixel_ratio.ToFloat()
-        viewport_size = QSizeF(aspectRatio, 1).scaled(window_size, Qt.KeepAspectRatio)
+        viewport_size = QSizeF(aspectRatio, 1).scaled(
+                            window_size, Qt.KeepAspectRatio
+                        ) * self.zoom
         viewport_rect = QRectF(QPointF(0, 0), viewport_size)
         viewport_rect.moveCenter(window_rect.center())
```

After that patch, it produces the same results as this code.

(Within a pixel one way or the other, attributable to the difference between passing floats to `QRect()` — which I _think_ in Python < 3.10 effectively called `int()` on all of the arguments, meaning it truncated floats — vs. using `QRectF.toRect()`, which will round to the nearest integer)

But also:
1. It doesn't break Python 3.10
2. It respects pixel aspect ratio. (_Shouldn't_ the viewport be doing that? I feel like it should. Unless it's already applied somewhere else.)